pontoon/base/models.py

```diff
@@ -1876,6 +1878,10 @@ class ResourceQuerySet(models.QuerySet):
     def asymmetric(self):
         return self.filter(format__in=Resource.ASYMMETRIC_FORMATS)
 
+    def obsolete(self, now):
+        self.update(obsolete=True, date_obsoleted=now)
+        Entity.objects.filter(resource__in=self).update(obsolete=True, date_obsoleted=now)
```

Comment:
It's probably a good idea to also add `entity.obsolete()` and `entities.obsolete()` methods, which would update these two fields.